src/switch/src/Switch.js

```diff
@@ -172,10 +69,27 @@ class Switch extends PureComponent {
       appearance,
       hasCheckIcon,
       defaultChecked,
-      ...props
-    } = this.props
+      ...rest
+    } = props
+    const [isChecked, setIsChecked] = useState(
+      checkedProps || defaultChecked || false
+    )
+
+    const theme = useTheme()
+
+    const handleChange = useCallback(
+      value => {
+        if (checkedProps) {
+          onChange(value)
+        } else {
+          setIsChecked(checked => !checked)
+          onChange(value)
+        }
+      },
+      [onChange]
+    )
```

Comment:
This is a bit weird that it is both... controlled and uncontrolled? (e.g. passing in `checked` but also.. state is internally handled?)

Reply:
Agreed! -- preserving old behavior 🤷‍♂️

Reply:
Should we break from it? (can do in a follow up PR, but add to the v5 roadmap tracker)

Reply:
Definitely should. Imo no sense in having an uncontrolled component, especially iven that other inputs are controlled 🤷‍♂️